buildpacks/nodejs-corepack/src/main.rs

```diff
@@ -37,60 +40,76 @@ impl Buildpack for CorepackBuildpack {
     fn detect(&self, context: DetectContext<Self>) -> libcnb::Result<DetectResult, Self::Error> {
         // Corepack requires the `packageManager` key from `package.json`.
         // This buildpack won't be detected without it.
-        let pkg_json_path = context.app_dir.join("package.json");
-        if pkg_json_path.exists() {
-            let pkg_json =
-                PackageJson::read(pkg_json_path).map_err(CorepackBuildpackError::PackageJson)?;
-            cfg::get_supported_package_manager(&pkg_json).map_or_else(
-                || DetectResultBuilder::fail().build(),
-                |pkg_mgr| {
-                    DetectResultBuilder::pass()
-                        .build_plan(
-                            BuildPlanBuilder::new()
-                                .requires("node")
-                                .requires(&pkg_mgr)
-                                .provides(pkg_mgr)
-                                .build(),
-                        )
-                        .build()
-                },
-            )
-        } else {
-            DetectResultBuilder::fail().build()
-        }
+        let tracer = init_tracer(format!("{}", context.buildpack_descriptor.buildpack.id));
```

Comment:
this is where i think tighter integration with `libcnb.rs` would be helpful. auto-initializing the tracer (or disabling it if opt-in/out can be configured) and pre-populating a root span with the buildpack id (and maybe some other useful default attributes).

Reply:
Yeah, totally agree! I want to trial it here before bringing it into libcnb.rs for now though.